src/peft/tuners/tuners_utils.py

```diff
@@ -299,6 +299,16 @@ def unmerge_adapter(self):
             if isinstance(module, BaseTunerLayer):
                 module.unmerge()
 
+    def _unloading_checks(self, adapter_names: List[str]):
+        adapters_to_consider = adapter_names or self.active_adapters
+        is_modules_to_save_available = any(
+            self.peft_config[adapter].modules_to_save is not None for adapter in adapters_to_consider
+        )
+        if is_modules_to_save_available and len(adapters_to_consider) > 1:
+            raise ValueError(
+                "models having `modules_to_save` specified in LoraConfig support only single adapter for unloading."
```

Comment:
I tried to come up with an error message that is a bit clearer, WDYT?

```suggestion
                "Cannot unload multiple adapters that specify `modules_to_save`."
```